### PR TITLE
Update React Query documentation

### PR DESCRIPTION
**Pull request by afoon**
[Jira](https://dp3.atlassian.net/browse/MB-15491)
- [ ] add more explanations
- [ ] Add more examples
- [ ] Change name back to React Query for discoverability

### CODE REVIEW

File: docs/frontend/guides/how-we-use-react-query.md [hunk 1]
```diff
@@ -143,16 +143,60 @@ const onSubmit = async () => {
...
+  mutationFn: functionToBeCalled ,
+    // First callback
+   onSuccess: () => { 
+       console.log('Where logic should be handled') 
+       queryClient.invalidateQueries({queryKey: [myKey]})
+       },
+   onError: () => {
+       console.log('Error handling');
+       }
+   });
```
Comment (rogeruiz) [inline docs/frontend/guides/how-we-use-react-query.md:163]: Can we make these references too? I'm concerned with the code copied here being out of date going forward.

Comment (afoon) [reply docs/frontend/guides/how-we-use-react-query.md:163]: The references were moved to the section below with added context.

**Comment by rogeruiz** [comment]
I'm gonna take a further look at this today and give my review. I like the intentions of these changes. This is a great start.

File: docs/frontend/guides/how-we-use-react-query.md [hunk 2]
```diff
@@ -143,16 +143,60 @@ const onSubmit = async () => {
 :::
 
  Callbacks maybe not fire as expected. To avoid that issue, logic should be handled in the `useMutation` callback which is called first. 
- ```jsx reference
-https://github.com/transcom/mymove/blob/main/src/pages/Office/EditShipmentDetails/EditShipmentDetails.jsx#L22-L30
+
+```js 
+ import {useQueryClient} from '@tanstack/react-query'
```
Comment (rogeruiz) [inline docs/frontend/guides/how-we-use-react-query.md:148]:
```suggestion
```js title="use-mutation-example.jsx"
import {useQueryClient} from '@tanstack/react-query'
```

File: docs/frontend/guides/how-we-use-react-query.md [hunk 3]
```diff
@@ -143,16 +143,60 @@ const onSubmit = async () => {
...
+       console.log('Error handling');
+       }
+   });
 ```
+
 UI changes should happen in the `mutate` callback which is called second after the `useMutation` callback. This is handled second so the mutation can complete. 
 
-```jsx reference
-https://github.com/transcom/mymove/blob/main/src/components/Office/ShipmentForm/ShipmentForm.jsx#L330-L339
+```js
```
Comment (rogeruiz) [inline docs/frontend/guides/how-we-use-react-query.md:168]:
```suggestion
```js title="second-callback-example.jsx"
```

**Review by rogeruiz — CHANGES_REQUESTED** [review]
❌ This PR needs changes ...

I've added some comments on what I think we need to change. Some minor things, but otherwise this looks good for now.

**Comment by rogeruiz** [comment]
Also this is unrelated to the review, but I've noticed a lot of space characters at the end of the line. The doesn't have an `.editorconfig` file, but I'll be adding one based on the `transcom/mymove` one in a new PR.

**Review by rogeruiz — APPROVED** [review]
lgtm 🌈

**Pull request merged**